jr-annotation-support/src/main/java/com/fasterxml/jackson/jr/annotationsupport/AnnotationBasedValueRWModifier.java

```diff
@@ -75,4 +97,115 @@ public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws
             return _valueType;
         }
     }
+
+    /**
+     * Serialize an enum using the {@link JsonValue} tagged method
+     */
+    private static class EnumJsonValueWriter implements ValueWriter {
+        private final Class<?> _valueType;
+        private final Method _jsonValueMethod;
+
+        private EnumJsonValueWriter(Class<?> _valueType, Method _jsonValueMethod) {
+            this._valueType = _valueType;
+            this._jsonValueMethod = _jsonValueMethod;
+        }
+
+        @Override
+        public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws IOException {
+            try {
+                context.writeValue(_jsonValueMethod.invoke(value));
+            } catch (Exception e) {
+                throw new IOException("Cannot call JsonValue method", e);
+            }
+        }
+
+        @Override
+        public Class<?> valueType() {
+            return _valueType;
+        }
+
+        /**
+         * Scan the methods of the enum to find a {@link JsonValue} tagged method
+         * which takes no parameters and returns a {@link String}
+         * @param type the type
+         * @return either a {@link EnumJsonValueWriter} to write this enum with, or <code>null</code> if no suitable
+         * method found
+         */
+        private static EnumJsonValueWriter of(Class<?> type) {
+            if (!isEnum(type)) {
+                return null;
+            }
+            return getJsonValueFunction(type, type);
+        }
+
+        private static EnumJsonValueWriter getJsonValueFunction(Class<?> type, Class<?> inspectionType) {
+            for (Method method : inspectionType.getDeclaredMethods()) {
+                if (method.getDeclaredAnnotation(JsonValue.class) != null &&
+                        method.getParameterCount() == 0 &&
+                        method.getReturnType().equals(String.class)) {
```

Comment:
Also, note that:

    @JsonValue(false)

means "disabled", i.e. NOT to be used. Possibly not relevant with jackson-jr as the only use is via mix-ins -- but in theory should check "annotation.getValue()" being `true`.

Reply:
I've written a test that demonstrates that one can add a `@JsonValue(false)` to a subclass override of an `@JsonValue(true)` method in order to disable that -  this means that you can turn off the `@JsonValue` behaviour, but it doesn't mean you can replace it with a different one.

Reply:
That sounds good -- as  I suggested flexibility with this setting has mostly value with Jackson mix-ins and less so without. But thank you for covering the case there is, it is good to handle intended behavior.